lib_bgp_data/roas_parser/roas_automator.py

```diff
@@ -0,0 +1,191 @@
+                                         ROAs_Table, clear_table=False)
+                        table.create_index()
+
+                        # Create a backup, if it is successful, overwrite the
+                        # existing backup
+                        if self._backup_table():
+                            self._overwrite()
+                            break
+
+                        # If the backup is unsuccessful, restore the table
+                        # to the last backup
+                        if os.path.exists(self.prev_backup):
+                            table.clear_table()
+                            self._restore_table()
+                            table.create_index()
+
+                        # Enters this block if there is no previous backup,
+                        # need to make the table empty so that data can tried
+                        # to be inserted again
+                        else:
+                            table.execute("DELETE FROM roas")
+                            
+                    # If after three attempts the backups are unsuccessful,
+                    # then just insert the new roas and don't backup.
+                    else:
+                        utils.clean_paths([self.path, self.csv_dir])
+                        utils.rows_to_db(roas, _csv_dir,
+                                         ROAs_Table, clear_table=False)
+                        table.create_index()
+
+                        # Raise a backup error
+                        raise BackupError()
+
+                except subprocess.CalledProcessError as e:
+                    time = utils.now()
+                    subject = f"Roas Backup Error at {time}"
+                    body = ("There was an error backing up the roas table. "
+                            "Below are the outputs\n"
+                            f"STD_OUT Message: {e.stdout}\n"
+                            f"STD_ERR Message: {e.stderr}")
+                    utils.send_email(subject, body)
+
+                except BackupError:
+                    time = utils.now()
+                    subject = f"Roas Backup Error at {time}"
+                    body = ("There was an error backing up the roas table. "
+                            "The backup file was not created correctly, so "
+                            "the previous backup was not overwritten. The "
+                            "pg_dump and pg_restore commands did not provide "
+                            "an error.")
+                    utils.send_email(subject, body)
+
+########################
+### Helper Functions ###
+########################
+
+    def _backup_table(self):
+        """This funtion will backup the table and check the backup"""
+
+        # Get the number of rows in the table
+        with ROAs_Table() as table:
+            count = table.get_count()
+            
+        # Make a temporary backup
+        cmd = (f"sudo -i -u postgres "
```

Comment:
can you move this into postgres.py and add an additional unit test for this? Section should be an input, the base class Parser has a default section that should be used instead of bgp